samples/show-scale-bar/src/main/java/com/esri/arcgismaps/sample/showscalebar/screens/ShowScaleBarScreen.kt

```diff
@@ -0,0 +1,245 @@
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.arcgismaps.UnitSystem
+import com.arcgismaps.toolkit.geoviewcompose.MapView
+import com.arcgismaps.toolkit.scalebar.Scalebar
+import com.arcgismaps.toolkit.scalebar.ScalebarStyle
+import com.esri.arcgismaps.sample.sampleslib.components.DropDownMenuBox
+import com.esri.arcgismaps.sample.sampleslib.components.MessageDialog
+import com.esri.arcgismaps.sample.sampleslib.components.SampleTopAppBar
+import com.esri.arcgismaps.sample.sampleslib.theme.SampleAppTheme
+import com.esri.arcgismaps.sample.showscalebar.components.ShowScaleBarViewModel
+import com.esri.arcgismaps.sample.showscalebar.components.durationToSeconds
+import kotlin.time.Duration
+import kotlin.time.DurationUnit
+import kotlin.time.toDuration
+
+/**
+ * Main screen layout for the sample app
+ */
+@Composable
+fun ShowScaleBarScreen(sampleName: String) {
+    val mapViewModel: ShowScaleBarViewModel = viewModel()
+    // Keep track of the currently selected scalebar
+    var currentScalebarStyle by remember { mutableStateOf(ScalebarStyle.AlternatingBar) }
+    var isScalebarDialogOptionsVisible by remember { mutableStateOf(false) }
+    var currentAutoHideDelay by remember { mutableStateOf(Duration.INFINITE) }
+    var isGeodeticCalculationsEnabled by remember { mutableStateOf(true) }
+    var currentScalebarUnitSystem by remember { mutableStateOf<UnitSystem>(UnitSystem.Imperial) }
+
+    Scaffold(
+        topBar = { SampleTopAppBar(title = sampleName) },
+        floatingActionButton = {
+            if (!isScalebarDialogOptionsVisible) {
+                FloatingActionButton(
+                    modifier = Modifier.padding(bottom = 36.dp, end = 12.dp),
+                    onClick = { isScalebarDialogOptionsVisible = true }
+                ) { Icon(Icons.Filled.Settings, contentDescription = "Show Scalebar Options") }
+            }
+        },
+        content = { paddingValues ->
+            Box(
+                modifier = Modifier
+                    .fillMaxSize()
+                    .padding(paddingValues)
+            ) {
+                MapView(
+                    modifier = Modifier.fillMaxSize(),
+                    arcGISMap = mapViewModel.arcGISMap,
+                    isAttributionBarVisible = false,
+                    onSpatialReferenceChanged = mapViewModel::updateSpacialReference,
+                    onUnitsPerDipChanged = mapViewModel::updateUnitsPerDip,
+                    onViewpointChangedForCenterAndScale = mapViewModel::updateViewpoint
+                )
+                Scalebar(
+                    modifier = Modifier
+                        .padding(horizontal = 25.dp, vertical = 50.dp)
+                        .align(Alignment.BottomStart),
+                    maxWidth = 300.dp,
+                    unitsPerDip = mapViewModel.unitsPerDip,
+                    viewpoint = mapViewModel.viewpoint,
+                    spatialReference = mapViewModel.spatialReference,
```

Comment:
@shubham7109 why are you updating spatialReference, viewpoint and unitsPerDip via the viewModel ?
why not use RememberMutableStateof in this screen itself? on orientation changes mapView retains(saves) the viewpoint and restores it. SpatialReference will get recreated too. I think we want the latest value for UnitsPerDips as well. If we are holding on to the last value in the viewModel, does that mean when the scalebar redraws on orientation change, it might not be accurate.

Reply:
With samples we typically let the users write their rotation handling code, but to answer the question, spatialReference, viewpoint, and unitsPerDip updates are triggered via the MapView and held by the MapViewModel. While testing rotations, the mapview does recompose, and tiggeres a new update for all spatialReference, viewpoint, and unitsPerDip.  Meaning, the Scalebar should consume the latest values even after recomposition.  

@puneet-pdx Let me know if I'm missing something, or we could discuss this on a call.